19-pied-raven/technical_specification.md

```diff
@@ -0,0 +1,64 @@
+2. Explore how to connect, store and retrieve secrets
+3. Provide a synchronous API for communication with the Vault
+
+### 2. Integrate with and adapt adjacent services
+Both the Encryption Key Store and the Interrogation Room depend on (synchronous and asynchronous) communication with other services.
+Interaction with the respective service on side of the caller/receiver is not implemented yet.
+Check and adjust all of the following services:
+
+1. Upload Controller - outbound event signaling file upload completed to Interrogation Room
+2. Upload Controller - handling of failure event from Interrogation Room
+3. Internal File Registry - handling of success event from Interrogation Room
+4. Download Controller - request and receive envelope, attach envelope to outgoing file cotent
+
+### 3. Finalize request/response models and incoming/outgoing events
+While models and events exist for all the necessary cross-service communication, some have a more prototypical character and might need updates to their definition based on spec compliance or practical concerns.
+Those updates should be applied after all services can communicate with each other.
+This includes moving (remaining) in-service event definitions to the https://github.com/ghga-de/ghga-event-schemas repository.
+
+### 4. Provide a local setup covering all services using docker-compose
+As all services dealing with file upload/download/encryption/decryption interact with each other, either during upload or download, a local setup to test functionality across all services would be beneficial.
+This would allow to capture issues early and replace mocks with actual data to see how well our microservice architecture performs.
+
+The implementation of this journey consists of providing a docker-compose file that yields a working local setup.
+A minimal test case ensuring functionality should be included.
+
+## Optional User Journeys:
+
+### 1. Decide on user public key propagation/storage/retrieval mechanism
+Currently a user's public key is piped through the interrogation room into the encryption key store.
+We need to discuss specifics of public key propagation and storage, as this permeates some of the models/events and issues like source of truth and expulsion of keys need to be resolved.
+However, this might be more fitting to consider in conjunction with a refactoring step for integration with the auth story.
+## User Journeys that are not part of this Epic:
+
+## API Definitions:
+
+### RESTful/Synchronous:
+[Encryption Key Store REST API](./api_definitions/rest/encryption_key_store.yml) - [Swagger UI](https://editor.swagger.io/?url=https://raw.githubusercontent.com/ghga-de/epic-docs/main/19-pied-raven/api_definitions/rest/encryption_key_store.yml) - This supersedes the 18 Hooded Crow version.
+
+## Additional Implementation Details:
+
+
+## Human Resource/Time Estimation:
+
+Number of sprints required: 2-3
+
+Number of developers required: 2
```

Comment:
Maybe rather 3 developers?

Reply:
With the reduced scope we can do it with 2 full devs in 2 full sprints. Since we don't have full devs and also don't have full sprints, we could keep the estimation or put 3 devs on it.